### PR TITLE
github action init

### PR DESCRIPTION
**Pull request by zacharyblasczyk**

### CODE REVIEW

File: integrations/fetch-job-inputs/package.json [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    "GitHub",
+    "Actions",
+    "Ctrlplane"
+  ],
+  "exports": {
+    ".": "./dist/index.js"
+  },
+  "packageManager": "pnpm@9.7.1",
+  "engines": {
+    "node": "20"
```
Comment (jsbroks) [inline integrations/fetch-job-inputs/package.json:15]: 22?

File: integrations/fetch-job-inputs/package.json [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  },
+  "type": "module",
+  "main": "index.ts",
+  "scripts": {
+    "dev": "pnpm exec tsx src/index.ts",
+    "build": "tsc",
```
Comment (jsbroks) [inline integrations/fetch-job-inputs/package.json:21]: we need nvc

File: integrations/fetch-job-inputs/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      const releaseVersion = response.release?.version;
+      const location = response.target?.config.location;
+      const project = response.target?.config.project;
+      const variables = response.variables;
+
+      core.setOutput("target_name", targetName);
+      core.setOutput("environment_name", environmentName);
+      core.setOutput("release_version", releaseVersion);
+      core.setOutput("target.location", location);
+      core.setOutput("target.project", project);
```
Comment (jsbroks) [inline integrations/fetch-job-inputs/src/index.ts:46]:
```suggestion
      core.setOutput("target.name", targetName);
      core.setOutput("environment.name", environmentName);
      core.setOutput("release.version", releaseVersion);
      core.setOutput("target.location", location);
      core.setOutput("target.project", project);
```

File: integrations/fetch-job-inputs/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  apiKey: core.getInput("api_key", { required: true }),
+});
+
+const api = new DefaultApi(config);
+
+async function fetchWithRetry(jobId: string) {
+  return pRetry(
+    async () => {
+      await api.acknowledgeJob({ jobId });
+      return await api.getJob({ jobId });
```
Comment (jsbroks) [inline integrations/fetch-job-inputs/src/index.ts:17]:
```suggestion
   api.getJob({ jobId })
```

File: integrations/fetch-job-inputs/src/index.ts (hunk 3 above)
Comment (jsbroks) [inline integrations/fetch-job-inputs/src/index.ts:46]:
```suggestion
      const { variables, target, release, environment } = response

      core.setOutput("target_name", targetName);
      core.setOutput("environment_name", environmentName);
      core.setOutput("release_version", releaseVersion);
      core.setOutput("target.location", location);
      core.setOutput("target.project", project);
```

File: integrations/fetch-job-inputs/src/index.ts [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        core.info(`${newPrefix}: ${String(value)}`);
+        core.setOutput(newPrefix, value);
+        console.log(`${newPrefix}: ${String(value)}`);
+      }
+    }
+  } else {
+    core.info(`${prefix}: ${String(obj)}`);
+    core.setOutput(prefix, obj);
+    console.log(`${prefix}: ${String(obj)}`);
+  }
```
Comment (jsbroks) [inline integrations/fetch-job-inputs/src/index.ts:32]: return early

File: packages/node-sdk/src/models/GetJob200Response.ts [hunk 6]
```diff
@@ -92,6 +92,12 @@ export interface GetJob200Response {
    * @memberof GetJob200Response
    */
   variables?: object;
+  /**
+   *
+   * @type {object}
+   * @memberof GetJob200Response
+   */
+  config: object;
```
Comment (jsbroks) [inline packages/node-sdk/src/models/GetJob200Response.ts:100]: no

File: integrations/fetch-job-inputs/package.json [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    "p-retry": "^6.2.0"
+  },
+  "devDependencies": {
+    "@ctrlplane/eslint-config": "workspace:*",
+    "@ctrlplane/prettier-config": "workspace:*",
+    "@ctrlplane/tsconfig": "workspace:*",
+    "@types/node": "catalog:node20",
+    "@vercel/ncc": "^0.38.1",
+    "eslint": "catalog:",
+    "tsx": "^4.19.1",
```
Comment (jsbroks) [inline integrations/fetch-job-inputs/package.json:38]: where do we use this

File: integrations/fetch-job-inputs/src/index.ts [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+const requiredOutputs = core
+  .getInput("required_outputs", { required: false })
+  .split("\n")
+  .filter((output) => output.trim() !== "")
+  .map((output) => output.trim());
+
+const outputTracker: Record<string, boolean> = {};
+const trackOutput = (key: string, value: any) => {
+  if (value !== undefined && value !== null) outputTracker[key] = true;
+};
```
Comment (jsbroks) [inline integrations/fetch-job-inputs/src/index.ts:21]: use a `new Set` add it if it exists, use `.has` to check

File: integrations/fetch-job-inputs/package.json (hunk 7 above)
Comment (zacharyblasczyk) [reply integrations/fetch-job-inputs/package.json:38]: Lemme move them to dev deps. Good catch.

File: integrations/fetch-job-inputs/src/index.ts [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+      setOutputsRecursively("target_config", target?.config);
+      setOutputsRecursively("variables", variables ?? {});
+    })
+    .then(() => {
+      if (requiredOutputs.length > 0) {
+        core.info(
+          `The required_outputs for this job are: ${requiredOutputs.join(", ")}`,
+        );
+
+        const missingOutputs = requiredOutputs.filter(
```
Comment (jsbroks) [inline integrations/fetch-job-inputs/src/index.ts:91]: check if every/some will work here

**Pull request merged**